### PR TITLE
Migrating basicpy to nf-test

### PR DESCRIPTION
**Pull request by kbestak**
Migrated testing of BasicPy to nf-test instead of pytest.

## PR checklist

Closes https://github.com/nf-core/modules/issues/6927

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements. --> not possible, conda not supported
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/basicpy/main.nf [hunk 1]
```diff
@@ -36,13 +36,17 @@ process BASICPY {
     if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
         error "Basicpy module does not support Conda. Please use Docker / Singularity instead."
     }
+    def args   = task.ext.args   ?: ''
```
Comment (SPPearce) [inline modules/nf-core/basicpy/main.nf:39]:
```suggestion
```

File: modules/nf-core/basicpy/tests/nextflow.config
Comment (SPPearce) [inline modules/nf-core/basicpy/tests/nextflow.config:1]: Can delete this file.

**Comment by SPPearce** [comment]
The tests are failing because the md5sums are not consistent, do they just need updating or do the files contain a datestamp or something?

**Comment by SPPearce** [comment]
@kbestak , are you able to continue here?

**Comment by kbestak** [comment]
Hi @SPPearce , I've opened a new PR now because I decided it would be easier than resolving conflicts: https://github.com/nf-core/modules/pull/7228

**Pull request closed without merging**